test/refactoring/ConvertAsync/convert_result.swift

```diff
@@ -1,513 +1,533 @@
+  // NESTEDBREAK-COMMENT-NEXT: // e
+  // NESTEDBREAK-COMMENT-NEXT: if test(str) {
+  // NESTEDBREAK-COMMENT-NEXT: // f
+  // NESTEDBREAK-COMMENT-NEXT:   <#break#>
+  // NESTEDBREAK-COMMENT-NEXT: // g
+  // NESTEDBREAK-COMMENT-NEXT: }
+  // NESTEDBREAK-COMMENT-NEXT: // h
+  // NESTEDBREAK-COMMENT-NEXT: print("result \(str)")
+  // NESTEDBREAK-COMMENT-NEXT: // i
+  // NESTEDBREAK-COMMENT-NEXT: // j
+  // NESTEDBREAK-COMMENT-NEXT: // k
+  // NESTEDBREAK-COMMENT-NEXT: // l
+  // NESTEDBREAK-COMMENT-NEXT: print("after")
+  // NESTEDBREAK-COMMENT-NEXT: // m
+  // NESTEDBREAK-COMMENT-NEXT: {{ }}
+  // NESTEDBREAK-COMMENT-NOT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=ERROR-BLOCK-COMMENT %s
+  simple { res in
+    // a
+    print("before")
+    // b
+    switch res {
+    case .success(let str):
+      // c
+      print("result \(str)")
+      // d
+    case .failure:
+      // e
+      print("fail")
+      // f
+      return
       // g
     }
     // h
-    print("result \(str)")
+    print("after")
     // i
-  case .failure:
-    // j
-    break
-    // k
   }
-  // l
-  print("after")
-  // m
-}
-// NESTEDBREAK-COMMENT:      let str = try await simple()
-// NESTEDBREAK-COMMENT-NEXT: // a
-// NESTEDBREAK-COMMENT-NEXT: // b
-// NESTEDBREAK-COMMENT-NEXT: print("before")
-// NESTEDBREAK-COMMENT-NEXT: // c
-// NESTEDBREAK-COMMENT-NEXT: // d
-// NESTEDBREAK-COMMENT-NEXT: // e
-// NESTEDBREAK-COMMENT-NEXT: if test(str) {
-// NESTEDBREAK-COMMENT-NEXT: // f
-// NESTEDBREAK-COMMENT-NEXT:   <#break#>
-// NESTEDBREAK-COMMENT-NEXT: // g
-// NESTEDBREAK-COMMENT-NEXT: }
-// NESTEDBREAK-COMMENT-NEXT: // h
-// NESTEDBREAK-COMMENT-NEXT: print("result \(str)")
-// NESTEDBREAK-COMMENT-NEXT: // i
-// NESTEDBREAK-COMMENT-NEXT: // j
-// NESTEDBREAK-COMMENT-NEXT: // k
-// NESTEDBREAK-COMMENT-NEXT: // l
-// NESTEDBREAK-COMMENT-NEXT: print("after")
-// NESTEDBREAK-COMMENT-NEXT: // m
-// NESTEDBREAK-COMMENT-EMPTY:
-// NESTEDBREAK-COMMENT-NOT: }
-
-// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ERROR-BLOCK-COMMENT %s
-simple { res in
-  // a
-  print("before")
-  // b
-  switch res {
-  case .success(let str):
-    // c
-    print("result \(str)")
-    // d
-  case .failure:
-    // e
-    print("fail")
-    // f
-    return
-    // g
+  // ERROR-BLOCK-COMMENT:      do {
+  // ERROR-BLOCK-COMMENT-NEXT:   let str = try await simple()
+  // ERROR-BLOCK-COMMENT-NEXT:   // a
+  // ERROR-BLOCK-COMMENT-NEXT:   print("before")
+  // ERROR-BLOCK-COMMENT-NEXT:   // b
+  // ERROR-BLOCK-COMMENT-NEXT:   // c
+  // ERROR-BLOCK-COMMENT-NEXT:   print("result \(str)")
+  // ERROR-BLOCK-COMMENT-NEXT:   // d
+  // ERROR-BLOCK-COMMENT-NEXT:   // h
+  // ERROR-BLOCK-COMMENT-NEXT:   print("after")
+  // ERROR-BLOCK-COMMENT-NEXT:   // i
+  // ERROR-BLOCK-COMMENT-NEXT:   {{ }}
```

Comment:
It's because we include the trailing trivia up until the next token, but in this case the next token gets removed by the transform, leaving a stray newline. At the time, it didn't seem like a case that would come up too often, given it should only happen for trailing comments at the end of braces. Though looking at it again, it seems like it should be pretty straightforward to fix, filed rdar://82072147 to take care of it. Thanks for pointing it out!